### PR TITLE
[MM-47566] Add upcominginvoice field to Subscription type

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
See: https://github.com/mattermost/customer-web-server/pull/1000

Adds support for UpcomingInvoice passed by CWS if parseable.

#### Release Note

```release-note
None
```

### CODE REVIEW

**Review by neallred — APPROVED** [review]

File: api4/cloud.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func (api *API) InitCloud() {
 	// GET /api/v4/cloud/subscription
 	api.BaseRoutes.Cloud.Handle("/subscription", api.APISessionRequired(getSubscription)).Methods("GET")
 	api.BaseRoutes.Cloud.Handle("/subscription/invoices", api.APISessionRequired(getInvoicesForSubscription)).Methods("GET")
-	api.BaseRoutes.Cloud.Handle("/subscription/invoices/{invoice_id:in_[A-Za-z0-9]+}/pdf", api.APISessionRequired(getSubscriptionInvoicePDF)).Methods("GET")
```
Comment (marianunez) [inline api4/cloud.go:40]: Is this intended?

File: web/context.go [hunk 2]
```diff
@@ -743,7 +743,7 @@ func (c *Context) RequireInvoiceId() *Context {
 		return c
 	}
 
-	if len(c.Params.InvoiceId) != 27 {
+	if len(c.Params.InvoiceId) != 27 && c.Params.InvoiceId != model.UpcomingInvoice {
```
Comment (marianunez) [inline web/context.go:746]: What are we testing here checking against `upcoming` in the `id` for get invoice requests?

Comment (marianunez) [reply web/context.go:746]: Nvm, saw your webapp PR now that requests the invoice with `upcoming` as Id

**Review by marianunez — APPROVED** [review]
Should've started with the webapp one first, my questions were answered there 👍 Thanks @nickmisasi!

**Pull request merged**